firefox_ui_harness/runtests.py

```diff
@@ -7,71 +7,57 @@
 import tempfile
 
 import mozinstall
-
-from mozlog import structured
+from marionette.runtests import MarionetteHarness, cli as mn_cli
 
 from firefox_ui_harness.arguments import FirefoxUIArguments
 from firefox_ui_harness.runners import FirefoxUITestRunner
 
 
-def startTestRunner(runner_class, options, tests):
-    install_folder = None
-
-    try:
-        # Prepare the workspace path so that all temporary data can be stored inside it.
-        if options.workspace_path:
-            path = os.path.expanduser(options.workspace_path)
-            options.workspace = os.path.abspath(path)
+class FirefoxUIHarness(MarionetteHarness):
+    def __init__(self,
+                 runner_class=FirefoxUITestRunner,
+                 parser_class=FirefoxUIArguments):
+        MarionetteHarness.__init__(self, runner_class, parser_class)
+        self.install_folder = None
 
-            if not os.path.exists(options.workspace):
-                os.makedirs(options.workspace)
-        else:
-            options.workspace = tempfile.mkdtemp('.{}'.format(os.path.basename(sys.argv[0])))
-
-        options.logger.info('Using workspace for temporary data: "{}"'.format(options.workspace))
+    def process_args(self):
+        # Make the default workspace TMP instead of .
+        # Note gecko.log and Fx profile will also be saved in workspace
+        # Note the workspace is not deleted after the test run
+        if not self.args.workspace:
+            self.args.workspace = tempfile.mkdtemp('.{}'.format(os.path.basename(sys.argv[0])))
```

Comment:
@whimboo Done. Good thing you mentioned this because it caused me to discover an omission in `UpdateTestRunner`.